install.sh

```diff
@@ -159,21 +268,21 @@ download_config() {
 # .env file or a default value.  Note that these variables do NOT affect the
 # docker containers.  They get their values strictly from the .env file.
 set_variables() {
-    log_info "Reading needed values from .env file."
+    log_info "Setting up .env configuration file."
 
-    MC_PORT="$DEFAULT_PORT"
-    if grep -q '^MC_PORT=' .env; then
-        MC_PORT=$(grep -E '^MC_PORT=' .env | cut -d'=' -f2-)
+    if grep -q '^#MC_PORT=' .env && [ "$HTTP_PORT" != "$DEFAULT_HTTP_PORT" ]; then
+        log_info "Using non-standard HTTP_PORT=${HTTP_PORT}"
+        portable_sed_i "s|^#MC_PORT=${DEFAULT_HTTP_PORT}|MC_PORT=${HTTP_PORT}|" .env
     fi
 
-    MC_SSL_PORT="$DEFAULT_SSL_PORT"
-    if grep -q '^MC_SSL_PORT=' .env; then
-        MC_SSL_PORT=$(grep -E '^MC_SSL_PORT=' .env | cut -d'=' -f2-)
+    if grep -q '^#MC_SSL_PORT=' .env && [ "$HTTPS_PORT" != "$DEFAULT_HTTPS_PORT" ]; then
+        log_info "Using non-standard HTTPS_PORT=${HTTPS_PORT}"
+        portable_sed_i "s|^#MC_SSL_PORT=${DEFAULT_HTTPS_PORT}|MC_SSL_PORT=${HTTPS_PORT}|" .env
     fi
 
-    MC_DEFAULT_XGT_PORT="$DEFAULT_XGT_PORT"
-    if grep -q '^MC_DEFAULT_XGT_PORT=' .env; then
-        MC_DEFAULT_XGT_PORT=$(grep -E '^MC_DEFAULT_XGT_PORT=' .env | cut -d'=' -f2-)
+    if grep -q '^#MC_XGT_PORT=' .env && [ "$XGT_PORT" != "$DEFAULT_XGT_PORT" ]; then
```

Comment:
I don't think there is a way to change the XGT PORT. The yaml has it hard coded:
https://github.com/Rocketgraphai/rocketgraph/blob/main/docker-compose.yml

Reply:
This needs to be fixed. Can you confirm it works as expected? I confirmed it didn't work in ps1 script, so left it out.